openapi.yaml

```diff
@@ -16,175 +16,301 @@ host: api.oregonstate.edu
         200:
           description: Successful response
           schema:
-            $ref: '#/definitions/PetsResult'
+            $ref: '#/definitions/NotesResult'
         400:
-          description: Bad Request
+          description: Bad request
           schema:
             $ref: '#/definitions/Errors'
         500:
-          description: Internal Server Error
+          description: Internal server error
           schema:
             $ref: '#/definitions/Errors'
-  /pets/{id}:
-    get:
-      summary: Info for a specific pet
+    post:
+      summary: Create a new note
       tags:
-        - pets
-      description: Get pet by unique ID
-      operationId: getPetById
+        - notes
+      operationId: postNotes
       parameters:
-        - $ref: '#/parameters/authorization'
-        - name: id
-          in: path
-          description: Unique Pet ID
-          required: true
-          type: string
+        - in: body
+          name: requestBody
+          schema:
+            $ref: '#/definitions/NoteResource'
+      responses:
+        200:
+          description: Successfully created a note
+          schema:
+            $ref: '#/definitions/NoteResult'
+        400:
+          description: Invalid note object
+          schema:
+            $ref: '#/definitions/Errors'
+        500:
+          description: Internal server error
+          schema:
+            $ref: '#/definitions/Errors'
+  /notes/{noteID}:
+    parameters:
+      - name: noteID
+        in: path
+        type: string
+        required: true
+    patch:
+      summary: Edit an existing note
+      tags:
+        - notes
+      operationId: patchNoteByID
+      parameters:
+        - in: body
+          name: requestBody
+          schema:
+            type: object
+            properties:
+              note:
+                $ref: '#/definitions/note'
+              permissions:
+                $ref: '#/definitions/permissions'
+              dateCreated:
+                $ref: '#/definitions/dateCreated'
+      responses:
+        200:
+          description: Successfully updated the note
+          schema:
+            $ref: '#/definitions/NoteResult'
+        400:
+          description: Invalid patch body
+          schema:
+            $ref: '#/definitions/Errors'
+        404:
+          description: Note not found
+          schema:
+            $ref: '#/definitions/Errors'
+        500:
+          description: Internal server error
+          schema:
+            $ref: '#/definitions/Errors'
+    delete:
+      summary: Delete an existing note
+      tags:
+        - notes
+      operationId: deleteNoteByID
+      responses:
+        204:
+          description: Successfully deleted the note
+        404:
+          description: Note not found
+          schema:
+            $ref: '#/definitions/Errors'
+  /notes/{noteID}/versions:
+    parameters:
+      - name: noteID
+        in: path
+        type: string
+        required: true
```

Comment:
`noteID` has appeared at least twice, so I think we can put it under `parameters` sections to make it reusable.